### ISSUE TITLE
How to run a UDF on vector cube to access geometry?

### ISSUE THREAD

**Issue by soxofaan** [Open-EO/openeo-processes#449]
We're trying to support/implement a use case that depends heavily on vector cube manipulation, in particular running some custom (Python) UDF on the geometry data.

For example some (GeoJSON) feature collection manipulation use cases that are straightforward to implement the classic way, outside the openEO framework:
- filter out features, based on geometry (area, location) and/or properties 
- add properties (in GeoJSON sense) to the features, based on geometry and or existing properties, using third party library, tool or service (e.g. lookup of population for a geometry or country code property)

How should these be implemented with openEO processes?
Note that with vector cubes, a lot of the relevant data to work with is in the labels (geometry data), which is quite different compared to how we typically work with raster data  cubes

**Comment by m-mohr** [comment]
I don't quite get the question @jdries:
- For x,y the labels are the x/y part respecively of the coordinate
- For t the labels are (usually) the RFC3339 timestamps
- For the bands the labels are the band names
- For the geometries we agreed on WKT of the geometry

See https://openeo.org/documentation/1.0/datacubes.html#dimensions for details/examples.

**Comment by jdries** [comment]
Yes, but the question is how to get to those labels from within a callback passed to apply_dimension:
> apply_dimension(cube,dimension='bands',process=my_callback)

This passes on a labeled array to my process, where the labels of the array will be band names, but what if I in addition need to know the single geometry label? (So the WKT string, for that labeled array.)

**Comment by m-mohr** [comment]
You could get all WKT strings for each array element using array_labels: https://processes.openeo.org/#array_labels

**Comment by m-mohr** [comment]
What single label of a geometry dimension? I don't get what you are looking for... Could you explain the use case a bit more?

**Comment by m-mohr** [comment]
@jdries This needs more input if we want to solve this issue for release v2.0.0-rc.2
